app/routes/document.js

```diff
@@ -15,4 +16,20 @@ export default class DocumentRoute extends Route {
       include: 'file',
     });
   }
+
+  async afterModel(model) {
+    const decisionActivity = await this.store.queryOne('decision-activity', {
+      filter: {
+        report: {
+          ':id:': model?.id,
+        },
+      },
+    });
+    this.pieceIsFromDecision = isPresent(decisionActivity);
```

Comment:
I would pass the decision-activity to the controller and apply this logic inside the controller